### PR TITLE
Remove deprecated angular table panel from SNMP mixin

### PR DESCRIPTION
**Pull request by mshahzeb**
Updates out-of-date angular table panel to new table panel.

Since the old grafonnet version does not have a new table panel - the dashboard has been updated in Grafana and saved as JSON.

image

### CODE REVIEW

File: snmp-mixin/dashboards/snmp-overview.json [hunk 1]
```diff
@@ -0,0 +1,1078 @@
...
+      },
+      "id": 2,
+      "panels": [],
+      "showTitle": true,
+      "targets": [
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "grafanacloud-prom"
+          },
```
Comment (Dasomeone) [inline snmp-mixin/dashboards/snmp-overview.json:58]: Minor nitpick, but pretty sure these are all meant to be no type and `"uid": "$datasource"`

File: snmp-mixin/dashboards/snmp-overview.json [hunk 2]
```diff
@@ -0,0 +1,1078 @@
...
+    "list": [
+      {
+        "current": {
+          "selected": false,
+          "text": "default",
+          "value": "default"
+        },
+        "hide": 0,
+        "includeAll": false,
+        "label": "Data Source",
```
Comment (Dasomeone) [inline snmp-mixin/dashboards/snmp-overview.json:934]: This won't pass linting, mixtool is getting quite particular on naming (e.g. should be `Data source`) Might be worth giving it a pass to see if there's anything else

**Review by Dasomeone — COMMENTED** [review]
Couple minor comments but otherwise looks great

File: snmp-mixin/dashboards/snmp-overview.json (hunk 2 above)
Comment (mshahzeb) [reply snmp-mixin/dashboards/snmp-overview.json:934]: Ran mixtool lint and it did not raise any issues.

Comment (Dasomeone) [reply snmp-mixin/dashboards/snmp-overview.json:934]: Interesting, then one of us has outdated mixtool :sweat_smile: 

Will update mine and test

File: snmp-mixin/dashboards/snmp-overview.json (hunk 1 above)
Comment (mshahzeb) [reply snmp-mixin/dashboards/snmp-overview.json:58]: This is actually a row panel and I dont think it even needs targets - removed it. Thanks for the catch!

File: snmp-mixin/dashboards/snmp-overview.json (hunk 2 above)
Comment (Dasomeone) [reply snmp-mixin/dashboards/snmp-overview.json:934]: Update: turns out it was my mixtool, so all good to go on this one :)

**Review by Dasomeone — APPROVED** [review]
LGTM, thanks Shahzeb :D

**Comment by rgeyer** [comment]
> Is there a replacement for the deleted `snmp-mixin/lib/utils.libsonnet` in this PR? For now I've copied the function.

The goal of this PR was to move away from using deprecated panels, which includes moving from graph to timeseries. That util function was a bit of a hack to patch that change in.

As such, now that the dashboard uses timeseries natively, the util isn't needed anymore (for this mixin). Were you using the function in some other project?

**Comment by zalegrala** [comment]
I was using the util for something else, but perhaps I need to migrate that as well.  In the meantime, the copy works fine for me while I figure out what direction to take the jsonnet dashbaords I've got.

**Pull request merged**